src/Sulu/Bundle/AdminBundle/Metadata/XmlParserTrait.php

```diff
@@ -16,7 +16,14 @@
  */
 trait XmlParserTrait
 {
-    private function getValueFromXPath($path, \DOMXPath $xpath, ?\DOMNode $context = null, $default = null)
+    /**
+     * @template T of mixed
+     *
+     * @param T $default
+     *
+     * @return T|bool|int|string|null
+     */
+    private function getValueFromXPath(string $path, \DOMXPath $xpath, ?\DOMNode $context = null, $default = null)
```

Comment:
improved the PHP Typing here the trait is already internal so needs not documentation in upgrade